File: kitten-house-care/src/main/java/org/jesperancinha/housing/data/CatDto.java
package org.jesperancinha.housing.data;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

import static java.util.List.of;

@Builder
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class CatDto {
    private String name;

    private String color;

    private String species;

    private String pattern;

    private Long age;

    private final List<OwnerDto> formerOwners = new ArrayList<>();

    private final List<CareCenterDto> careCenters = new ArrayList<>();
}

File: kitten-house-care/src/main/kotlin/org/jesperancinha/housing/data/CatDto.kt
package org.jesperancinha.housing.data

import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Getter
import lombok.NoArgsConstructor

@Builder
@Getter
@NoArgsConstructor
@AllArgsConstructor
class CatDto {
    private val name: String? = null
    private val color: String? = null
    private val species: String? = null
    private val pattern: String? = null
    private val age: Long? = null
    private val formerOwners: List<OwnerDto> = ArrayList()
    private val careCenters: List<CareCenterDto> = ArrayList()
}